### PR TITLE
Create release

### PR DESCRIPTION
**Pull request by outscale-vbr**
* Add default values
* Add githubci for release

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    - name: gh login
+      run: make gh-login
+      env:
+        SECRET_GITHUB_TOKEN: ${{ secrets.SECRET_GITHUB_TOKEN }}
+    - name: generate changelog
+      run: make release-changelog
+      env:
+        IMG_RELEASE: ${{ secrets.DOCKER_REGISTRY_REPO }}/cluster-api-outscale-controllers:${{ env.RELEASE_TAG }}
+        RELEASE_TAG: ${{ env.RELEASE_TAG }}
+        PREVIOUS_RELEASE_TAG: ${{ secrets.PREVIOUS_RELEASE_TAG }}
```
Comment (outscale-mdr) [inline .github/workflows/release.yml:46]: You would have to update the secret every release, could we not compute this ?

File: Makefile [hunk 2]
```diff
@@ -193,6 +198,85 @@ else
 	$(KUSTOMIZE) build config/default | $(ENVSUBST)  | kubectl delete --ignore-not-found=$(ignore-not-found) -f -
 endif
 
+##@ Release
+
+RELEASE_DIR := out
+GH_ORG_NAME ?= outscale-dev
+GH_REPO_NAME ?= cluster-api-provider-outscale
+RELEASE_BINARY ?= cluster-api
+GH_REPO ?= outscale-vbr/$(GH_REPO_NAME)
```
Comment (outscale-mdr) [inline Makefile:207]: change to outscale-dev

File: hack/releasechangelog.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    esac
+done
+
+echo "# Release notes for Cluster API Provider Outscale (CAPO) $RELEASE_TAG"
+echo "# Changelog since $PREVIOUS_RELEASE_TAG"
+
+if [ $PREVIOUS_RELEASE_TAG == "None" ]; then 
+	$GH api repos/$GH_ORG_NAME/$GH_REPO_NAME/releases/generate-notes -F tag_name=$RELEASE_TAG -f target_commitish='release-test' --jq '.body'
+else 
+        $GH api repos/$GH_ORG_NAME/$GH_REPO_NAME/releases/generate-notes -F tag_name=$RELEASE_TAG -F previous_tag_name=$PREVIOUS_RELEASE_TAG --jq '.body'
```
Comment (outscale-mdr) [inline hack/releasechangelog.sh:35]: indent

File: templates/cluster-template.yaml [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+          cloud-provider: external
+          provider-id: osc://'{{ ds.meta_data.instance_id }}'
+        name: '{{ ds.meta_data.local_hostname }}'
+    files:
+    - content: |
+        #!/bin/bash
+
+        curl https://github.com/opencontainers/runc/releases/download/v1.1.1/runc.amd64 -Lo /tmp/runc.amd64
+        chmod +x /tmp/runc.amd64	
+        \cp -f /tmp/runc.amd64 /usr/local/sbin/runc
```
Comment (outscale-mdr) [inline templates/cluster-template.yaml:163]: Is the '\' wanted ?

**Review by outscale-mdr — CHANGES_REQUESTED** [review]

File: templates/cluster-template.yaml (hunk 4 above)
Comment (outscale-vbr) [reply templates/cluster-template.yaml]: Change

File: hack/releasechangelog.sh (hunk 3 above)
Comment (outscale-vbr) [reply hack/releasechangelog.sh]: Done

File: Makefile (hunk 2 above)
Comment (outscale-vbr) [reply Makefile]: Done

File: .github/workflows/release.yml (hunk 1 above)
Comment (outscale-vbr) [reply .github/workflows/release.yml]: Done

**Comment by outscale-vbr** [comment]
Templates are for release

**Comment by outscale-vbr** [comment]
crd folder are generated

**Comment by outscale-vbr** [comment]
generate-notes  https://cli.github.com/manual/gh_release_create

**Review by outscale-hmi — COMMENTED** [review]
seems good for me

**Review by outscale-hmi — APPROVED** [review]
seems good for me

**Review by outscale-mdr — APPROVED** [review]

**Comment by outscale-vbr** [comment]
Close  https://github.com/outscale-dev/cluster-api-provider-outscale/issues/131

**Pull request merged**